client/src/document/ingredients/browser-compatibility-table/utils.ts

```diff
@@ -31,11 +31,31 @@ interface Feature {
   depth: number;
 }
 
+function findAllCompatDepths(
+  identifier: BCD.Identifier,
+  depths: number[],
+  flattenedKey: string
+) {
+  for (let key in identifier) {
+    if (identifier.hasOwnProperty(key)) {
+      if (key === "__compat") {
+        depths.push(flattenedKey ? flattenedKey.split(".").length : 0);
+      }
+      const newKey = flattenedKey ? flattenedKey + "." + key : key;
+      const subIdentifier = identifier[key];
+      if (typeof subIdentifier === "object") {
+        findAllCompatDepths(subIdentifier as BCD.Identifier, depths, newKey);
+      }
+    }
+  }
+}
```

Comment:
Let's have this return the depth directly, allowing us to make this more efficient with breath-first search (BFS):

```suggestion
function findFirstCompatDepth(
  identifier: BCD.Identifier
) {
  const entries = [
    ["", identifier]
  ];

  while (entries.length) {
    const [path, value] = entries.shift() as [string, BCD.Identifier];

    if (value.hasOwnProperty("__compat")) {
      // Following entries have at least this depth.
      return path.split(".").length;
    }

    for (const key of Object.keys(value)) {
      entries.push([path + "." + key, value[key]]);
    }
  }

  // Fallback.
  return 0;
}
```

Reply:
I tried to figure out something like this, but I couldn't figure it out. I just tried this provided code, and got the same issue I kept getting when I tried to write something like this myself: http://localhost:3000/en-US/docs/Mozilla/Add-ons/WebExtensions/API/devtools#browser_compatibility and http://localhost:3000/en-US/docs/Mozilla/Add-ons/WebExtensions/API/privacy#browser_compatibility have no entries with this code.

Reply:
This shows http://localhost:3000/en-US/docs/Mozilla/Add-ons/WebExtensions/API/devtools/panels#browser_compatibility incorrectly.

Reply:
Reason is as mentioned here: https://github.com/mdn/yari/issues/5041#issuecomment-1249377330

Reply:
I don't doubt there is a better way to do this, but I had trouble thinking of it, and this does not work correctly.

Reply:
In BFS, the invariant holds that following entries have at least the same depth.

But I did indeed have an off-by-one error, because all recursive paths would start with a `.`.

This should do it:

```suggestion
function findFirstCompatDepth(
  identifier: BCD.Identifier
) {
  const entries = [
    ["", identifier]
  ];

  while (entries.length) {
    const [path, value] = entries.shift() as [string, BCD.Identifier];

    if (value.hasOwnProperty("__compat")) {
      // Following entries have at least this depth.
      return path.split(".").length;
    }

    for (const key of Object.keys(value)) {
      const subpath = path ? `${path}.${key}` : key;
      entries.push([subpath, value[key]]);
    }
  }

  // Fallback.
  return 0;
}
```